lib/new_relic/agent/instrumentation/rack/instrumentation.rb

```diff
@@ -11,6 +11,7 @@ class << builder_class
             attr_accessor :_nr_deferred_detection_ran
           end
           builder_class._nr_deferred_detection_ran = false
+          NewRelic::Control::SecurityInterface.instance.wait = true
```

Comment:
Are these changes to make sure the Rack instrumentation continues to load in a deferred way, even when the security agent is present?

Reply:
Yes. I'd love to know more about how well our deferred behavior works with modern Ruby apps to know if it itself has any shortcomings or if there's anything we're doing that we don't need to do anymore in 2023.

Reply:
Great idea. Captured in https://github.com/newrelic/newrelic-ruby-agent/issues/2046